Assets/PlayerAssets/Scripts/ClickToMove.cs

```diff
@@ -5,21 +5,45 @@
 
 public class ClickToMove : MonoBehaviour {
 	// public GameObject playerBodyRef;
-	NavMeshAgent playerNavMeshAgent;
-	public LayerMask movementMask;
+	NavMeshAgent playerNavMeshAgent;					//Reference to the player
+	Transform target;									//Target for following
+	float stoppingDistanceMultiplier;					//Multiply this by the radius of the object to get a stopping point for interacting
 
-	// Use this for initialization
 	void Start () {
-		
+		stoppingDistanceMultiplier = 0.8f;
 		playerNavMeshAgent = transform.parent.GetComponent<NavMeshAgent>();
 	}
 
+	///Move the player
+	public void movePlayer(Vector3 destination) {
+		playerNavMeshAgent.SetDestination(destination);
+	}
+
+	///Move to an object
+	public void moveToObject(Interactable destinationObject) {
+		playerNavMeshAgent.stoppingDistance = destinationObject.radius * stoppingDistanceMultiplier;
+		target = destinationObject.interactionTransform;
+		playerNavMeshAgent.SetDestination(target.position);
+	}
+	///Follow the target
+	public void FollowTarget(Interactable newTarget) {
```

Comment:
Just a note, we are going to have to be careful of how much we use coroutines. I would want a check to see if the character is actually a moving object then call this function if it is, not every time we right click an object. Keep that in mind.

Reply:
Yeah i agree. Just so you know, the way the code is now, that coroutine will never run. It's there in case we decide we want to constantly follow something